mojo/stdlib/stdlib/collections/list.mojo

```diff
@@ -837,6 +837,54 @@ struct List[T: Copyable & Movable, hint_trivial_type: Bool = False](
 
         return ret_val^
 
+    fn pop(mut self, slice: Slice) -> Self:
+        """Pops a slice of items from the list.
+
+        Args:
+            slice: The slice of items to pop.
+
+        Returns:
+            A list of popped items.
+        """
```

Comment:
Curious, is the reason this is returning a new `List` so this can be parallel with the `pop(index)` function? Is it useful to return the new list or should it be omitted?